py/selenium/webdriver/remote/webdriver.py

```diff
@@ -1222,3 +1226,112 @@ def delete_downloadable_files(self) -> None:
             raise WebDriverException("You must enable downloads in order to work with downloadable files.")
 
         self.execute(Command.DELETE_DOWNLOADABLE_FILES)
+
+    @property
+    def fedcm(self) -> FedCM:
+        """
+        :Returns:
+            - FedCM: an object providing access to all Federated Credential Management (FedCM) dialog commands.
+
+        :Usage:
+            ::
+
+                title = driver.fedcm.title
+                subtitle = driver.fedcm.subtitle
+                dialog_type = driver.fedcm.dialog_type
+                accounts = driver.fedcm.account_list
+                driver.fedcm.select_account(0)
+                driver.fedcm.click_continue()
+                driver.fedcm.cancel_dialog()
+                driver.fedcm.enable_delay()
+                driver.fedcm.disable_delay()
+                driver.fedcm.reset_cooldown()
+        """
+        return self._fedcm
+
+    @property
+    def supports_fedcm(self) -> bool:
+        """Returns whether the browser supports FedCM capabilities."""
+        return self.capabilities.get(ArgOptions.FEDCM_CAPABILITY, False)
+
+    def _require_fedcm_support(self):
+        """Raises an exception if FedCM is not supported."""
+        if not self.supports_fedcm:
+            raise WebDriverException(
+                "This browser does not support Federated Credential Management. "
+                "Please ensure you're using a supported browser."
+            )
+
+    @property
+    def dialog(self):
+        """Returns the FedCM dialog object for interaction."""
+        self._require_fedcm_support()
+        return Dialog(self)
+
+    def enable_fedcm_delay(self):
```

Comment:
Why do we have this if we have `driver.fedcm.delay`?

Reply:
They are redundant now, will remove them from `webdriver.py`- `enable_fedcm_delay`, `disable_fedcm_delay`, `reset_fedcm_cooldown`.